code/game/objects/structures/platforms.dm

```diff
@@ -1,156 +1,174 @@
-	name = "platform"
-	desc = "A square metal surface resting on four legs."
-	icon = 'icons/obj/structures/platforms.dmi'
-	icon_state = "platform_deco"
-	anchored = TRUE
-	density = FALSE
-	layer = 3.5
-	flags_atom = ON_BORDER
-	resistance_flags = UNACIDABLE
 
-/obj/structure/platform_decoration/Initialize(mapload)
+	AddElement(/datum/element/connect_loc, (platform_flags & ENTRY_SLOWDOWN) ? (platform_flags & EXIT_SLOWDOWN) ? entry_and_exit : entry_only : exit_only)
+
+/obj/structure/platform/update_overlays()
 	. = ..()
-	switch(dir)
-		if (NORTH)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTH)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTHEAST)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTHWEST)
-			layer = ABOVE_MOB_LAYER
+	var/image/new_overlay
+
+	if(dir & EAST)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, EAST)
+		new_overlay.pixel_x = 32
+		. += new_overlay
+
+	if(dir & WEST)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, WEST)
+		new_overlay.pixel_x = -32
+		. += new_overlay
+
+	if(dir & NORTH)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTH)
+		new_overlay.pixel_y = 32
+		new_overlay.layer = ABOVE_MOB_LAYER //perspective
+		. += new_overlay
+
+	if(dir & SOUTH)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTH)
+		new_overlay.pixel_y = -32
+		. += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, NORTHEAST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTHEAST)
+		new_overlay.pixel_y = 32
+		new_overlay.pixel_x = 32
+		new_overlay.layer = ABOVE_MOB_PLATFORM_LAYER
+		. += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, NORTHWEST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTHWEST)
+		new_overlay.pixel_y = 32
+		new_overlay.pixel_x = -32
+		new_overlay.layer = ABOVE_MOB_PLATFORM_LAYER
+		. += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, SOUTHEAST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTHEAST)
+		new_overlay.pixel_y = -32
+		new_overlay.pixel_x = 32
+		. += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, SOUTHWEST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTHWEST)
+		new_overlay.pixel_y = -32
+		new_overlay.pixel_x = -32
+		. += new_overlay
+
+///Applies slowdown when entering if applicable
+/obj/structure/platform/proc/on_enter(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	SIGNAL_HANDLER
+	if(!ismob(arrived))
+		return
+	if(arrived.throwing)
+		return NONE
+	if((arrived.status_flags & INCORPOREAL))
+		return
+	if(!isturf(old_loc))
+		return
+	if(get_dist(src, old_loc) != 1) //teleporter shenanigans
+		return
+
+	var/mob/arriving_mob = arrived
+	var/old_loc_dir = get_dir(src, old_loc)
+	if(old_loc_dir & dir)
+		arriving_mob.client.move_delay += climb_slowdown //applied directly to client movedelay as mob delay would only apply after an additional movement
```

Comment:
I dont get why you backload the slowdown instead of a quick doafter while crossing which is logically more consistant

Reply:
Remember the autoclimb PR? Its a good concept but in practice it was really annoying.